test/common/common/log_verbosity_update_test.cc

```diff
@@ -0,0 +1,229 @@
+  const char* file_3 = "envoy/baz.cc";
+  std::atomic<spdlog::logger*> flogger{nullptr};
+  getFancyContext().initFancyLogger(file_1, flogger);
+  getFancyContext().initFancyLogger(file_2, flogger);
+  getFancyContext().initFancyLogger(file_3, flogger);
+
+  std::pair<absl::string_view, int> update_1 = std::make_pair("envoy/*", 3);
+  getFancyContext().updateVerbositySetting({update_1});
+
+  SpdLoggerSharedPtr p = getFancyContext().getFancyLogEntry(file_1);
+  EXPECT_NE(p, nullptr);
+  EXPECT_EQ(p->level(), spdlog::level::warn);
+  p = getFancyContext().getFancyLogEntry(file_2);
+  EXPECT_NE(p, nullptr);
+  EXPECT_EQ(p->level(), spdlog::level::warn);
+  p = getFancyContext().getFancyLogEntry(file_3);
+  EXPECT_NE(p, nullptr);
+  EXPECT_EQ(p->level(), spdlog::level::warn);
+
+  std::pair<absl::string_view, int> update_2 = std::make_pair("envoy/src/*", 2);
+  getFancyContext().updateVerbositySetting({update_2});
+
+  p = getFancyContext().getFancyLogEntry(file_1);
+  EXPECT_EQ(p->level(), spdlog::level::info);
+  p = getFancyContext().getFancyLogEntry(file_2);
+  EXPECT_EQ(p->level(), spdlog::level::info);
+  p = getFancyContext().getFancyLogEntry(file_3);
+  EXPECT_EQ(p->level(), Logger::Context::getFancyDefaultLevel());
+
+  std::pair<absl::string_view, int> update_3 = std::make_pair("*/src/*", 4);
+  getFancyContext().updateVerbositySetting({update_3});
+
+  p = getFancyContext().getFancyLogEntry(file_1);
+  EXPECT_EQ(p->level(), spdlog::level::err);
+  p = getFancyContext().getFancyLogEntry(file_2);
+  EXPECT_EQ(p->level(), spdlog::level::err);
+  p = getFancyContext().getFancyLogEntry(file_3);
+  EXPECT_EQ(p->level(), Logger::Context::getFancyDefaultLevel());
+
+  std::pair<absl::string_view, int> update_4 = std::make_pair("*", 2);
+  getFancyContext().updateVerbositySetting({update_4});
+
+  p = getFancyContext().getFancyLogEntry(file_1);
+  EXPECT_EQ(p->level(), spdlog::level::info);
+  p = getFancyContext().getFancyLogEntry(file_2);
+  EXPECT_EQ(p->level(), spdlog::level::info);
+  p = getFancyContext().getFancyLogEntry(file_3);
+  EXPECT_EQ(p->level(), spdlog::level::info);
+
+  std::pair<absl::string_view, int> update_5 = std::make_pair("/*", 1);
+  getFancyContext().updateVerbositySetting({update_5});
+
+  p = getFancyContext().getFancyLogEntry(file_1);
+  EXPECT_EQ(p->level(), Logger::Context::getFancyDefaultLevel());
+  p = getFancyContext().getFancyLogEntry(file_2);
+  EXPECT_EQ(p->level(), Logger::Context::getFancyDefaultLevel());
+  p = getFancyContext().getFancyLogEntry(file_3);
+  EXPECT_EQ(p->level(), Logger::Context::getFancyDefaultLevel());
+}
+
+TEST(Fancy, globQuestionUpdate) {
+  const char* file_1 = "envoy/src/foo.cc";
+  const char* file_2 = "envoy/f__.cc";
+  const char* file_3 = "/bar.cc";
+  std::atomic<spdlog::logger*> flogger{nullptr};
+  getFancyContext().initFancyLogger(file_1, flogger);
+  getFancyContext().initFancyLogger(file_2, flogger);
+  getFancyContext().initFancyLogger(file_3, flogger);
```

Comment:
are these not clobbering the same flogger?

Reply:
Nope. actually this can be static I am thinking, and we still stored the underlying loggers.